### ISSUE TITLE
How do I reset ID?

### ISSUE THREAD

**Issue by overflowz** [OptimalBits/bull#564]
Well, I'm trying to implement repeatable job by myself (cron is not an option for me, because jobs should be ran every < 30 seconds).

Question is, ID is growing and growing (even though I don't need it, because I'm using custom jobId) and eventually, it will cause redis to error after max value of int64.

Here's the code I'm using.

````
...

QWatcher.on('completed', () => {
  resetWatcher();
});

async function resetWatcher() {
  try {
    const existingJob: bull.Job = await QWatcher.getJob('watcher');
    if (existingJob && (existingJob as any).failedReason) {
      await QWatcher.empty();
    }

    await (QWatcher as any).add({}, {
      jobId: 'watcher',
      removeOnComplete: true,
      removeOnFail: true,
      delay: 5000
    });
  } catch (err) {
    console.error(`watcher:resetWatcher ${err}`);
  }
};

resetWatcher();
````

Is this correct approach? If not, how should I repeat jobs every let's say, 5 seconds and not get overflow error from redis if max value of int64 reached?

**Comment by manast** [comment]
the repeat functionality relies on custom job ids internally, so it is not possible to use custom job ids with repeat. However, regarding the redis error due to overflow, how do you reproduce that?

**Comment by overflowz** [comment]
````
127.0.0.1:6379> set mykey 9223372036854775807
OK
127.0.0.1:6379> incr mykey
(error) ERR increment or decrement would overflow
````

**Comment by manast** [comment]
ok, but you mean that potentially the repeat functionality could overflow it? The custom job id generated by bull is: ```var customId = 'repeat:' + name + ':' + nextMillis;``` so it is not an int but a string. Not sure I understand in which case the overflow you mentioned happens.

**Comment by overflowz** [comment]
Yes it does, but bull:jobname:id still get's incremented and incremented. When it reaches maximum value, it will overflow the id. I haven't seen how bull handles it.

**Comment by manast** [comment]
aha, now I understand what you mean. I guess I will need to write a test case for that. uint64 is still a humongous value, should never happen in practice.

**Comment by overflowz** [comment]
Sorry, I was really tired and thought it would overflow in weeks. But anyways, resetting the ID (or handling the overflow) would be great.

Feel free to close the topic, I'll still wait for the answer if it's possible though.
